lib/graph3d/Settings.js

```diff
@@ -467,6 +469,50 @@ function setDataColor(dataColor, dst) {
   }
 }
 
+/**
+ * Converts an object that accepts a HUE keyword. This is converted to a certain amount of hex color stops. At which point:
+ * the HTML hex color codes is converted into an RGB color object.
+ * 
+ * @param {Object | Array<string>} surfaceColors Either an object that describes the HUE, or an array of HTML hex color codes
+ * @param {Object} dst 
+ */
+function setSurfaceColor(surfaceColors, dst) {
+  if(surfaceColors === undefined) {
+    return;    // Nothing to do
+  }
+
+  if (dst.surfaceColors === undefined) {
+    dst.surfaceColors = {};
+  }
+
+  let rgbColors = [];
+
+  if(Array.isArray(surfaceColors)) {
+    rgbColors = surfaceColors.map(util.hexToRGB);
+  } else if (typeof surfaceColors === 'object') {
+    if(surfaceColors.hue === undefined) {
+      throw new Error('Unsupported type of surfaceColors');
+    }
+
+    const startHue = surfaceColors.hue.start;
+    const endHue = surfaceColors.hue.end;
```

Comment:
I would use something like:
```javascript
const startHue = Math.min(surfaceColors.hue.start, surfaceColors.hue.end);
const endHue = Math.max(surfaceColors.hue.start, surfaceColors.hue.end);
```
and after the `rgbColors` are filled:
```javascript
if (surfaceColors.hue.start > surfaceColors.hue.end) {
  rgbColors.reverse();
}
```
This will allow for hues to go the other way around instead of ignoring the options and using the default. For example from blue through green to red.

Reply:
If the start hue is greater than end hue, then the hue will decrease from start to end.